platform/pulpcore/app/tasks/repository.py

```diff
@@ -9,8 +10,27 @@ def delete(repo_name):
     """
     Delete a :class:`~pulpcore.app.models.Repository`
 
-    :param repo_name:       the name of a repository
-    :type  repo_name:       str
+    Args:
+        repo_name (str): The name of the repository to be deleted
     """
 
     models.Repository.objects.filter(name=repo_name).delete()
+
+
+@shared_task(base=UserFacingTask)
+def update(repo_name, partial=True, data=None):
```

Comment:
s/repo_name/repository_id/

If the user changes the name of the repo in another, the id will still be the same.